twindb_backup/source/mysql_source.py

```diff
@@ -1,32 +1,101 @@
+# -*- coding: utf-8 -*-
+"""
+Module defines MySQL source class for backing up local MySQL.
+"""
 import os
-import shlex
 import tempfile
 import time
 
 from ConfigParser import NoOptionError
 from contextlib import contextmanager
-from pymysql.err import InternalError
 from subprocess import Popen, PIPE
-from twindb_backup import log, get_files_to_delete
+
+import pymysql
+
+from twindb_backup import LOG, get_files_to_delete, INTERVALS
 from twindb_backup.source.base_source import BaseSource
-from twindb_backup.util import get_connection
 
 
 class MySQLSourceError(Exception):
     """Errors during backups"""
 
 
+class MySQLConnectInfo(object):  # pylint: disable=too-few-public-methods
+    """MySQL connection details """
+    def __init__(self, defaults_file,
+                 connect_timeout=10,
+                 cursor=pymysql.cursors.DictCursor):
+
+        self.cursor = cursor
+        self.connect_timeout = connect_timeout
+        self.defaults_file = defaults_file
+
+
 class MySQLSource(BaseSource):
-    def __init__(self, defaults_file, run_type, config, dst):
-        self.defaults = defaults_file
-        self._suffix = 'xbstream.gz'
+    """MySQLSource class"""
+    def __init__(self, mysql_connect_info, run_type, full_backup, dst):
+        """
+        MySQLSource constructor
+
+        :param mysql_connect_info: MySQL connection details
+        :type mysql_connect_info: MySQLConnectInfo
+        :param run_type:
+        :param full_backup: When to do full backup e.g. daily, weekly
+        :type full_backup: str
+        :param dst:
+        """
+
+        class _BackupInfo(object):  # pylint: disable=too-few-public-methods
```

Comment:
We should set this as a schematics type in the next diff. Though its trivial to do it even in this one.
Why is this class defined inside the `init` function? I would think that this would be usable outside of the class as well. Or may be further down in the class, so doesn't make sense to define it inside `init`.